eng/references.targets

```diff
@@ -40,12 +40,12 @@
           AfterTargets="AssignProjectConfiguration"
           BeforeTargets="PrepareProjectReferences"
           Condition="'$(TargetFrameworkIdentifier)' == '.NETCoreApp' and
-                     ('$(IsTestProject)' == 'true' or '$(IsTestSupportProject)' == 'true') and
-                     '@(ProjectReferenceWithConfiguration)' != ''">
+                     ('$(IsTestProject)' == 'true' or '$(IsTestSupportProject)' == 'true' or ('$(IsSrcProject)' == 'true' and '$(IsPackable)' == 'true')) and
```

Comment:
I doubt that IsSrcProject exists. You probably mean IsSourceProject? Apart from that, why do you condition that on IsPackable only? I think we should do this consistently for all source projects, not just for the ones that are packable.

Reply:
i did for packaging projects to reduce side effects and do it where we need it.

Reply:
IMO we should apply this on all source projects, otherwise it's very hard to reason about when projects are transitive, vs when they aren't.